frontend/public/module/k8s/types.ts

```diff
@@ -46,6 +46,7 @@ export type K8sResourceKind = K8sResourceCommon & {
   };
   status?: { [key: string]: any };
   data?: { [key: string]: any };
+  headers?: { [key: string]: any };
```

Comment:
You're using the wrong type for the HTTP response. `K8sResourceKind` represents the resource itself, not an HTTP response. You should have a separate type for the response.

Reply:
@spadgett Are you referring to `headers?: { [key: string]: any };`?  If so, this is just for `headers` only not the entire response.

Reply:
Right, but that's not part of a k8s resource itself. Only part of the HTTP response in API requests, correct?

Reply:
@spadgett Based on the extended implementation of `consoleFetchCommon` to return the entire HTTP response, the `response.headers` is part of the k8s resource. Reason: we have to resolve the `response.body` before accessing the `response.headers` else we'll be displaying a warning policy alert for  a failed request which should not be the case, right?. See edit-yaml.jsx for details.

Reply:
So we're mixing in HTTP headers and response codes into a k8s resource object? That seems bad. It would be better to return essentially what `fetch` does and have a separate property for the body that includes the resource.

I think we should revert the public API and give this more thought since it's not something we can easily change later. These functions are really basic and fundamental to the API. I'm in favor of exposing HTTP status and headers for plugins that need it, but we want a clean, easy-to-understand API and that is also backwards compatible.

Reply:
@spadgett I think I see your concern. The `headers` property in `K8sResourceKind` type  unnecessary and is not used in any way. Sorry, that was my mistake.  Also, maybe I did not explain well earlier, let me try to explain better what I meant by "part of k8 resource" above. I mean the  response `headers` are returned along with the `body` for the `k8sCreate` action which is used for creating a resource. If you take a look at the edit-yaml.yaml line 337 where the entire response is consumed, you see that we are retrieving the `data` which is actually the `K8sResource` and the `headers` object too. So the `headers` is not part of ``K8sResource` .

Let me know if this makes sense now or you want me to set up a call so I can explain more.

Reply:
> @spadgett I think I see your concern. The `headers` property in `K8sResourceKind` type is unnecessary. Sorry, that was my mistake. Also, maybe I did not explain well earlier, let me try to explain better what I meant by "part of k8 resource" above. I mean the response `headers` are returned along with the `body` for the `k8sCreate` action which is used for creating a resource. If you take a look at the edit-yaml.yaml line 337 where the entire response is consumed, you see that we are retrieving the `data` which is actually the `K8sResource` and the `headers` object too. So the `headers` is not part of ``K8sResource` .
> 
> Let me know if this makes sense now or you want me to set up a call so I can explain more.

@spadgett Do you want me to set up a call to explain more?

Reply:
> > @spadgett I think I see your concern. The `headers` property in `K8sResourceKind` type is unnecessary. Sorry, that was my mistake. Also, maybe I did not explain well earlier, let me try to explain better what I meant by "part of k8 resource" above. I mean the response `headers` are returned along with the `body` for the `k8sCreate` action which is used for creating a resource. If you take a look at the edit-yaml.yaml line 337 where the entire response is consumed, you see that we are retrieving the `data` which is actually the `K8sResource` and the `headers` object too. So the `headers` is not part of ``K8sResource` .
> > Let me know if this makes sense now or you want me to set up a call so I can explain more.
> 
> @spadgett Do you want me to set up a call to explain more?

Also, I have removed the  unused `headers` property in `K8sResourceKind ` and updated the PR. PTAL